skrub/_agg_joiner.py

```diff
@@ -0,0 +1,597 @@
+        self-aggregation on the input data.
+
+    foreign_key : str or iterable of str
+        Select the columns from the auxiliary dataframe to use as keys during
+        the join operation.
+
+    main_key : str or iterable of str
+        Select the columns from the main table to use as keys during
+        the join operation.
+        If main_key is a list, we will perform a multi-column join.
+
+    cols : str or iterable of str, default=None
+        Select the columns from the auxiliary dataframe to use as values during
+        the aggregation operations.
+        If None, cols are all columns from table, except `foreign_key`.
+
+    operation : str or iterable of str, default=None
+        Aggregation operations to perform on the auxiliary table.
+
+        numerical : {"sum", "mean", "std", "min", "max", "hist", "value_counts"}
+            'hist' and 'value_counts' accepts an integer argument to parametrize
+            the binning.
+
+        categorical : {"mode", "count", "value_counts"}
+
+        If set to None (the default), ['mean', 'mode'] will be used.
+
+    suffix : str or iterable of str, default=None
+        The suffixes that will be add to each table columns in case of
+        duplicate column names, similar to `("_x", "_y")` in :obj:`~pandas.merge`.
+        If set to None, we will use the table index, by looking at their
+        order in 'tables', e.g. for a duplicate columns: price (main table),
+        price_1 (auxiliary table 1), price_2 (auxiliary table 2), etc.
+
+    See Also
+    --------
+    AggTarget :
+        Aggregates the target `y` before joining its aggregation
+        on the base dataframe.
+
+    Joiner :
+        Augments a main table by automatically joining multiple
+        auxiliary tables on it.
+
+    Examples
+    --------
+    >>> main = pd.DataFrame({
+            "airportId": [1, 2],
+            "airportName": ["Paris CDG", "NY JFK"],
+        })
+    >>> aux = pd.DataFrame({
+            "flightId": range(1, 7),
+            "from_airport": [1, 1, 1, 2, 2, 2],
+            "total_passengers": [90, 120, 100, 70, 80, 90],
+            "company": ["DL", "AF", "AF", "DL", "DL", "TR"],
+        })
+    >>> join_agg = AggJoiner(
+            table=aux,
+            foreign_key="from_airport",
+            main_key="airportId",
+            cols=["total_passengers", "company"],
+            operation=["mean", "mode"],
+        )
+    >>> join_agg.fit_transform(main)
+        airportId airportName company_mode  total_passengers_mean
+    0          1   Paris CDG           AF             103.333333
+    1          2      NY JFK           DL              80.000000
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        table: DataFrameLike | Iterable[DataFrameLike] | str | Iterable[str],
+        *,
+        foreign_key: str | Iterable[str],
+        main_key: str | Iterable[str],
+        cols: str | Iterable[str] | None = None,
```

Comment:
I think Iterable[Iterable[str]] is also allowed to specify a list of columns for each table? this is a more general discussion but it seems the type annotations in skrub in general don't type check: `mypy  skrub`  returns many errors. shall we consider either fixing the annotations and adding mypy to the CI, or removing the annotations?

Reply:
> this is a more general discussion but it seems the type annotations in skrub in general don't type check.

I agree, this is a discussion we need to have. It feels redundant to have both type hints in the function signatures and parameters in the docstrings.